src/witan/models/dem/ccm/components_functions.clj

```diff
@@ -78,3 +68,80 @@
                                vec
                                (sort-by (apply juxt col-key))
                                ds/dataset)))
+
+(defn lag
+  [dataset col-key]
+  (-> dataset
+      (ds/to-map)
+      (col-key)
+      (#(cons 0 %))
+      (butlast)))
+
+(defn calc-proportional-differences
+  "Takes a dataset of projected rates and returns a dataset with the proportional
+   difference in the rate between each year and the previous year. Also takes first
+   and last years of projection, and keyword with the scenario :low, :principal or :high."
+  [future-rates-trend-assumption first-proj-year last-proj-year scenario]
+  (-> future-rates-trend-assumption
+      (i/query-dataset {:year {:$gte first-proj-year :$lte last-proj-year}})
+      (ds/select-columns [:sex :age :year scenario])
+      (ds/rename-columns {scenario :national-trend})
+      (order-ds [:sex :age :year])
+      (#(ds/add-column % :last-year-nt (lag % :national-trend)))
+      (wds/add-derived-column :age [:age] inc)
+      (wds/add-derived-column :prop-diff [:national-trend :last-year-nt :year]
+                              (fn [national-trend last-year-nt year]
+                                (if (== year first-proj-year)
+                                  0
+                                  (wds/safe-divide [(- national-trend last-year-nt) last-year-nt]))))
+      (ds/select-columns [:sex :age :year :prop-diff])))
+
+(defn apply-national-trend
+  "Takes a dataset of projected rates for the jumpoff year, and a dataset of
+   projected national rates for the following years of the projection (currently
+   this is ONS data). Takes parameters for first & last projection years, the
+   scenario to be used from the future rates dataset (e.g. :low, :principal, or 
+   :high), and the name of the column with rates in the jumpoff year rates dataset
+   (e.g. :death-rate). Returns a dataset with the trend from the projected national
+   rates applied to the data from the jumpoff year dataset, for all of the year of
+   the projection."
+  [projected-rates-jumpoffyr future-rates-trend-assumption
+   first-proj-yr last-proj-yr scenario trend-column]
```

Comment:
@elisehuard- my understanding from the diagram [here] (https://github.com/MastodonC/witan.models.demography/blob/master/doc/images/Component_projection_alternatives.png) is that if the national trend is applied to values instead of rates, then the dataset being returned would be different- the schema would require :deaths instead of :death-rate, for the mortality example. So I think we'd need another function completely- @sunnytownsend?

And yes, good point about the column name- I find the use of 'trend' to be somewhat confusing in general. Will change to `assumption-col`.